feat: add `math/base/special/sech`

MSP20086 opened the pull request:
Resolves #228 

## Description

> What is the purpose of this pull request?

This pull request:

-   Computes the value of hyperbolic secant of an input in radian
-   It also includes the C-implementation

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #228 
-   fixes #

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

<img width="310" alt="image" src="https://github.com/stdlib-js/stdlib/assets/129577900/9682938e-ee53-4fd0-a9e4-3d35ff91770d">
<img width="589" alt="image" src="https://github.com/stdlib-js/stdlib/assets/129577900/64213018-968b-4d5b-8c22-441bf57e2ae7">
<img width="524" alt="image" src="https://github.com/stdlib-js/stdlib/assets/129577900/3c9b868f-31f3-4d85-af4e-957a67b49337">


## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

MSP20086 committed:
feat: add math/base/special/sech

Pranavchiku commented on lib/node_modules/@stdlib/math/base/special/sech/lib/main.js:
```diff
@@ -0,0 +1,57 @@
+* var v = sech( NaN );
+* // returns NaN
+*/
+function sech( x ) {
+	return 1.0 / cosh( x );
+}
```
@kgryte @Planeshifter 

Is there any other way to implement this, or we can proceed this way?

kgryte replied:
This is fine for now and is as proposed in the original RFC.

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/README.md:
````diff
@@ -0,0 +1,109 @@
+var sech = require( '@stdlib/math/base/special/sech' );
+```
+
+## csc( x )
````
```suggestion
## sech( x )
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/README.md:
```diff
@@ -0,0 +1,109 @@
+
+# Hyperbolic Secant
+
+> Evaluate the [hyperbolic secant][trigonometric-functions] of a number.
```
```suggestion
> Evaluate the [hyperbolic secant][trigonometric-functions] of a double-precision floating-point number.
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/README.md:
```diff
@@ -0,0 +1,109 @@
+
+## csc( x )
+
+Evaluates the [hyperbolic secant][trigonometric-functions] of `x` (in radians).
```
```suggestion
Evaluates the [hyperbolic secant][trigonometric-functions] of `x`.
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/README.md:
```diff
@@ -0,0 +1,109 @@
+var linspace = require( '@stdlib/array/base/linspace' );
+var sech = require( '@stdlib/math/base/special/sech' );
+
+var x = linspace( -10, 10, 100 );
```
```suggestion
var x = linspace( -10.0, 10.0, 100 );
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/README.md:
```diff
@@ -0,0 +1,109 @@
+
+<section class="related">
+
+* * *
+
+## See Also
+
+<span class="package-name">[`@stdlib/math/base/special/cosh`][@stdlib/math/base/special/cosh]</span><span class="delimiter">: </span><span class="description">evaluate the hyperbolic cosine of a number.</span>
```
```suggestion
```

As mentioned in the comment, we don't manually populate this section.

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/README.md:
```diff
@@ -0,0 +1,109 @@
+
+[trigonometric-functions]: https://en.wikipedia.org/wiki/Trigonometric_functions
+
+<!-- <related-links> -->
+
+[@stdlib/math/base/special/cosh]: https://github.com/stdlib-js/math/tree/main/base/special/cosh
+
+<!-- </related-links> -->
+
```
```suggestion
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/benchmark/benchmark.js:
```diff
@@ -0,0 +1,73 @@
+	b.pass( 'benchmark finished' );
+	b.end();
+});
+
+bench(pkg + '::built-in', function benchmark( b ) {
+	var x;
+	var y;
+	var i;
+
+	b.tic();
+	for ( i = 0; i < b.iterations; i++ ) {
+		x = ( randu() * 20.0 ) - 10.0;
+		y = 1.0 / cosh( x );
+		if ( isnan( y ) ) {
+			b.fail('should not return NaN');
+		}
+	}
+	b.toc();
+	if ( isnan( y ) ) {
+		b.fail( 'should not return NaN' );
+	}
+	b.pass( 'benchmark finished' );
+	b.end();
+});
```
```suggestion
```

This benchmark is not applicable here.

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/benchmark/benchmark.js:
```diff
@@ -0,0 +1,73 @@
+var bench = require( '@stdlib/bench' );
+var randu = require( '@stdlib/random/base/randu' );
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var cosh = require( '@stdlib/math/base/special/cosh' );
```
```suggestion
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/benchmark/benchmark.native.js:
```diff
@@ -0,0 +1,58 @@
+	b.tic();
+	for ( i = 0; i < b.iterations; i++ ) {
+		x = ( randu() * 10.0 ) - 5.0;
+		y = sech(x);
```
```suggestion
		y = sech( x );
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/benchmark/benchmark.js:
```diff
@@ -0,0 +1,73 @@
+	b.tic();
+	for ( i = 0; i < b.iterations; i++ ) {
+		x = ( randu() * 20.0 ) - 10.0;
+		y = sech(x);
```
```suggestion
		y = sech( x );
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/benchmark/c/Makefile:
```diff
@@ -0,0 +1,107 @@
+c_targets := benchmark.out
+
+
+# TARGETS #
```
Replace the contents of this file with an updated version: https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/abs/benchmark/c/Makefile

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,145 @@
+*/
+
+/**
+ * Benchmark `tanh`.
```
```suggestion
 * Benchmark `sech`.
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,145 @@
+    int i;
+
+    t = tic();
+    for (i = 0; i < ITERATIONS; i++)
```
TAB indentation.

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,145 @@
+        printf("ok %d benchmark finished\n", i + 1);
+    }
+    print_summary(REPEATS, REPEATS);
+}
```
Trailing newlines here and in every other file in this PR.

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/docs/repl.txt:
```diff
@@ -0,0 +1,28 @@
+
+{{alias}}( x )
+    Computes the hyperbolic secant of a number.
```
```suggestion
    Computes the hyperbolic secant of a double-precision floating-point number.
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/examples/c/example.c:
```diff
@@ -0,0 +1,33 @@
+
+int main(void)
+{
+    const double x[] = {-4.0, -3.11, -2.22, -1.33, -0.44, 0.44, 1.33, 2.22, 3.11, 4.0};
```
```suggestion
    const double x[] = { -4.0, -3.11, -2.22, -1.33, -0.44, 0.44, 1.33, 2.22, 3.11, 4.0 };
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/examples/c/example.c:
```diff
@@ -0,0 +1,33 @@
+{
+    const double x[] = {-4.0, -3.11, -2.22, -1.33, -0.44, 0.44, 1.33, 2.22, 3.11, 4.0};
+
+    double v;
```
TAB indentation and trailing newline.

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/examples/c/example.c:
```diff
@@ -0,0 +1,33 @@
+
+    double v;
+    int i;
+    for (i = 0; i < 10; i++)
+    {
```
```suggestion
    for ( i = 0; i < 10; i++ ) {
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/examples/index.js:
```diff
@@ -0,0 +1,29 @@
+var linspace = require( '@stdlib/array/base/linspace' );
+var sech = require( './../lib' );
+
+var x = linspace( -5.0, 5.0, 100 );
```
This should match the README.

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/include/stdlib/math/base/special/sech.h:
```diff
@@ -0,0 +1,39 @@
+{
+#endif
+
+    /**
+     * Computes the hyperbolic secant of a double-precision floating-point number.
+     */
+    double stdlib_base_sech(const double x);
```
```suggestion
/**
* Computes the hyperbolic secant of a double-precision floating-point number.
*/
double stdlib_base_sech( const double x );
```

Adhere to project conventions.

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/lib/index.js:
```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+/**
+* Evaluate the hyperbolic secant of a number.
```
```suggestion
* Evaluate the hyperbolic secant of a double-precision floating-point number.
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/lib/index.js:
```diff
@@ -0,0 +1,50 @@
+* // returns NaN
+*/
+
+
```
```suggestion
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/package.json:
```diff
@@ -0,0 +1,70 @@
+{
+  "name": "@stdlib/math/base/special/sech",
+  "version": "0.0.0",
+  "description": "Compute the hyperbolic secant of a number.",
```
```suggestion
  "description": "Compute the hyperbolic secant of a double-precision floating-point number.",
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/package.json:
```diff
@@ -0,0 +1,70 @@
+    "cosec",
+    "csc",
+    "trig",
+    "trigonometry",
+    "radians",
+    "angle"
```
```suggestion
    "trigonometry",
    "hyperbolic"
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/src/addon.c:
```diff
@@ -0,0 +1,23 @@
+#include "stdlib/math/base/napi/unary.h"
+
+// cppcheck-suppress shadowFunction
+STDLIB_MATH_BASE_NAPI_MODULE_D_D(stdlib_base_sech)
```
```suggestion
STDLIB_MATH_BASE_NAPI_MODULE_D_D( stdlib_base_sech )
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/src/main.c:
````diff
@@ -0,0 +1,54 @@
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+*
+* ## Notice
+*
+* The original C code, long comment, copyright, license, and constants are from [Cephes]{@link http://www.netlib.org/cephes}. The implementation follows the original, but has been modified according to project conventions.
+*
+* ```text
+* Copyright 1984, 1995, 2000 by Stephen L. Moshier
+*
+* Some software in this archive may be from the book _Methods and Programs for Mathematical Functions_ (Prentice-Hall or Simon & Schuster International, 1989) or from the Cephes Mathematical Library, a commercial product. In either event, it is copyrighted by the author. What you see here may be used freely but it comes with no support or guarantee.
+*
+* Stephen L. Moshier
+* moshier@na-net.ornl.gov
+* ```
````
```suggestion
```

This license comment is not applicable.

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/src/main.c:
````diff
@@ -0,0 +1,54 @@
+* ```
+*/
+
+#include "stdlib/math/base/special/cosh.h"
+#include "stdlib/math/base/special/sech.h"
````
```suggestion
#include "stdlib/math/base/special/sech.h"
#include "stdlib/math/base/special/cosh.h"
```

Include the package header before others.

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/src/main.c:
```diff
@@ -0,0 +1,54 @@
+#include "stdlib/math/base/special/cosh.h"
+#include "stdlib/math/base/special/sech.h"
+
+/* End auto-generated functions. */
+
```
```suggestion
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/src/main.c:
```diff
@@ -0,0 +1,54 @@
+/* End auto-generated functions. */
+
+/**
+ * Computes the hyperbolic secant of a double-precision floating-point number.
```
The indentation here is incorrect.

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/src/main.c:
```diff
@@ -0,0 +1,54 @@
+ */
+double stdlib_base_sech(const double x)
+{
+    double z;
+    z = stdlib_base_cosh(x);
+    z = 1 / z;
+    return z;
```
```suggestion
	return 1.0 / stdlib_base_cosh( x );
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/src/main.c:
```diff
@@ -0,0 +1,54 @@
+ * double out = stdlib_base_sech( 0.0 );
+ * // returns 1.0
+ */
+double stdlib_base_sech(const double x)
+{
```
```suggestion
double stdlib_base_sech( const double x ) {
```

Follow project conventions.

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/test/fixtures/julia/runner.jl:
```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env julia
```
Did you actually run this file or did you copy-paste the fixture JSON files from a different package?

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/test/test.native.js:
```diff
@@ -0,0 +1,198 @@
+
+// MODULES //
+
+var tape = require('tape');
+var isnan = require('@stdlib/math/base/assert/is-nan');
+var isPositiveZero = require('@stdlib/math/base/assert/is-positive-zero');
+var PINF = require('@stdlib/constants/float64/pinf');
+var NINF = require('@stdlib/constants/float64/ninf');
+var EPS = require('@stdlib/constants/float64/eps');
+var abs = require('@stdlib/math/base/special/abs');
+var sech = require('./../lib/native.js');
```
Follow project conventions wrt spacing.

kgryte replied:
That applies here and everywhere in this file.

kgryte commented on lib/node_modules/@stdlib/math/base/special/sech/test/test.native.js:
```diff
@@ -0,0 +1,198 @@
+
+// MODULES //
+
+var tape = require('tape');
+var isnan = require('@stdlib/math/base/assert/is-nan');
+var isPositiveZero = require('@stdlib/math/base/assert/is-positive-zero');
+var PINF = require('@stdlib/constants/float64/pinf');
+var NINF = require('@stdlib/constants/float64/ninf');
+var EPS = require('@stdlib/constants/float64/eps');
+var abs = require('@stdlib/math/base/special/abs');
+var sech = require('./../lib/native.js');
```
You cannot assume that you can load `./lib/native.js`. See other `math/base/special/*` packages with C implementations and emulate how they handle native add-on tests.

kgryte reviewed (changes requested):
Thanks @MSP20086 for working on this. At the moment, this PR needs considerable work before it is ready for merge. I've left several comments. In general, I suggest spending some time closing examining existing packages having C implementations and more closely matching the conventions found in those packages.

MSP20086 commented:
@kgryte Sorry for not following the conventions, will pay close attention to it in future. I will surely try to solve all the issues, according to suggestions.

kgryte commented:
@MSP20086 Are you still interested in working on this PR?

kgryte commented:
/stdlib merge

kgryte commented:
@gunjjoshi Do you think you'd be interested in taking over this PR?

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into develop2

gunjjoshi commented:
@kgryte Sure, I'll work on this!

gunjjoshi committed:
feat: add math/base/special/sech

kgryte commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into develop2

kgryte commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

kgryte commented:
Looks like this PR has some indentation errors.

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/sech/src/main.c:
```diff
@@ -0,0 +1,38 @@
+* // returns ~0.2658
+*/
+double stdlib_base_sech( const double x ) {
+    return 1.0 / stdlib_base_cosh( x );
```
```suggestion
	return 1.0 / stdlib_base_cosh( x );
```

gunjjoshi committed:
Update lib/node_modules/@stdlib/math/base/special/sech/src/main.c

Signed-off-by: Gunj Joshi <gunjjoshi8372@gmail.com>

gunjjoshi committed:
style: fix spacing

Signed-off-by: Gunj Joshi <gunjjoshi8372@gmail.com>

gunjjoshi committed:
style: replace spaces by tab

Signed-off-by: Gunj Joshi <gunjjoshi8372@gmail.com>

gunjjoshi committed:
style: remove trailing whitespace

Signed-off-by: Gunj Joshi <gunjjoshi8372@gmail.com>

anandkaranubc commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into develop2

anandkaranubc committed:
docs: fix example value

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
docs: replace manual for loop in examples

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
docs: fix redundant line break

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
docs: add relevant keywords

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
docs: remove additional example for consistency

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
bench: clean-up and add julia benchmarks

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
test: update tests for consistency with cosh

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc reviewed (dismissed):
Good to go!

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/sech/src/addon.c:
```diff
@@ -0,0 +1,23 @@
+#include "stdlib/math/base/special/sech.h"
+#include "stdlib/math/base/napi/unary.h"
+
+// cppcheck-suppress shadowFunction
+STDLIB_MATH_BASE_NAPI_MODULE_D_D( stdlib_base_sech )
```
```suggestion
STDLIB_MATH_BASE_NAPI_MODULE_D_D( stdlib_base_sech )
```

anandkaranubc committed:
docs: re-enable cppcheck lint rule

Signed-off-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
